### ISSUE TITLE
False positive: EOL comment may not be preceded by a block comment

### ISSUE THREAD

**Issue by igorwojda** [pinterest/ktlint#2599]
## Expected Behavior
Black line is there so rule should pass

## Observed Behavior
```
Step 'ktlint' found problem in 'src/integrationTest/kotlin/com/lemonappdev/konsist/core/declaration/koclass/KoClassDeclarationForKoParentClassProviderTest.kt':
Error on line: 124, column: 28
rule: standard:no-consecutive-comments
an EOL comment may not be preceded by a block comment unless separated by a blank line
java.lang.AssertionError: Error on line: 124, column: 28
rule: standard:no-consecutive-comments
an EOL comment may not be preceded by a block comment unless separated by a blank line
```

## Steps to Reproduce
```kotlin
@Test
    fun `object-has-parent-class-with-duplicated-name`() {
        /*
        Some comment
        */

        // given
		val a = 2
}
```

## Your Environment

* Version of ktlint used: `1.2.1`
* Spotless

```kotlin
spotless {
    kotlin {
        project.fileTree(project.rootDir) {
            exclude(".gradle/**")
        }

        ktlint("1.2.1")

        indentWithSpaces()
        endWithNewline()
    }

checks
    isEnforceCheck = false
}
```

* `com.diffplug.spotless:6.25.0`
* Version of Gradle used (if applicable): Gradle `8.6`
* Operating System and version: MacOs Sonoma

**Comment by igorwojda** [comment]
Woks as expected
